Converters/ArcGIS/Speckle.Converters.ArcGIS3/Utils/LocalToGlobalConverterUtils.cs

```diff
@@ -30,100 +31,82 @@ public Base TransformObjects(Base atomicObject, List<Matrix4x4> matrix)
     {
-      return TransformMesh(mesh, matrix);
+      if (listVals.ToList().FindAll(x => x is not ITransformable).Count == 0)
+      {
+        return TransformObjWithDisplayValues(atomicObject, transforms);
+      }
+      throw new SpeckleConversionException(
+        $"Display Values of types '{listVals.ToList().FindAll(x => x is not ITransformable).Select(y => y.speckle_type).Distinct().ToList()}' for {atomicObject.speckle_type} are not supported for local to global coordinate transformation"
+      );
     }
-    throw new SpeckleException(
-      $"{newObject.speckle_type} is not supported for local to global coordinate transformation"
+    throw new SpeckleConversionException(
+      $"{atomicObject.speckle_type} is not supported for local to global coordinate transformation"
     );
   }
 
-  private Base TransformObjWithDisplayValues(Base baseObj, List<Matrix4x4> matrix)
+  private Base TransformObjWithDisplayValues(Base atomicObject, List<Objects.Other.Transform> transforms)
   {
-    Base newObject = Core.Api.Operations.Deserialize(Core.Api.Operations.Serialize(baseObj));
-    List<SOG.Mesh> newDisplayValue = new();
+    // for all objects that are not transformable, but contain displayValue
+    List<Base> newDisplayValues = new();
 
-    var displayValue = newObject.TryGetDisplayValue();
+    var displayValue = atomicObject.TryGetDisplayValue();
     if (displayValue is null) // will not happen due to the check in "TransformObjects"
     {
-      throw new SpeckleException($"{newObject.speckle_type} blocks contains no display value");
+      throw new SpeckleConversionException($"{atomicObject.speckle_type} blocks contains no display value");
     }
 
     foreach (Base displayVal in displayValue)
     {
-      if (displayVal is SOG.Mesh displayMesh)
+      if (displayVal is ITransformable c)
       {
-        var newMesh = TransformMesh(displayMesh, matrix);
-        newDisplayValue.Add(newMesh);
+        foreach (var transform in transforms)
+        {
+          c.TransformTo(transform, out ITransformable newObj);
+          c = newObj;
+        }
+        newDisplayValues.Add((Base)c);
       }
       else // will not happen due to the check in "TransformObjects"
       {
-        throw new SpeckleException($"Blocks containing {baseObj.speckle_type} are not supported");
+        throw new SpeckleConversionException(
+          $"Blocks containing {displayVal.speckle_type} as displayValue are not supported"
+        );
       }
     }
-
-    newObject["displayValue"] = newDisplayValue;
-    return newObject;
-  }
-
-  private SOG.Mesh TransformMesh(SOG.Mesh displayMesh, List<Matrix4x4> matrix)
-  {
-    List<List<double>> oldVertices = new();
-    for (int i = 0; i < displayMesh.vertices.Count; i += 3)
+    // copy the original object and assign new displayValue - hacky
+    Base newObject = Core.Api.Operations.Deserialize(Core.Api.Operations.Serialize(atomicObject));
```

Comment:
This is creating a tight coupling to serialization we should avoid. Once, object versioning is in this is unlikely yo be viable - I would argue that it is beyond the scope of this method, probably this should become a service and we should avoid calling the static directly